packages/shared/src/hooks/usePostModalNavigation.ts

```diff
@@ -87,20 +87,17 @@ export const usePostModalNavigation = (
     onChangeSelected(index, fromPopState);
   };
 
-  const onCloseModal = useCallback(
-    (fromPopState = false) => {
-      setOpenedPostIndex(null);
-      setCurrentPage(undefined);
-      if (!fromPopState) {
-        window.scrollTo(0, scrollPositionOnFeed.current);
-
-        changeHistory({}, `Feed`, currentPage);
-      }
+  const onCloseModal = useCallback((fromPopState = false) => {
+    setOpenedPostIndex(null);
+    setCurrentPage(undefined);
+    if (!fromPopState) {
+      window.scrollTo(0, scrollPositionOnFeed.current);
 
-      scrollPositionOnFeed.current = 0;
-    },
-    [changeHistory, currentPage],
-  );
+      window.history.back();
```

Comment:
While this is not wrong, it won't be able to cover an edge case where the user tried to navigate using arrow keys left or right (or the navigation on the header) since going back would essentially be going from the previous post. To reproduce try to:
- Open a post modal.
- Hit the left or right arrow key on your keyboard.
- Close the modal.
- Inspect the browser URL.
- The path should just be `/`.